pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -44,16 +44,13 @@ def __init__(self) -> None:
         self._num_dataloaders: Optional[int] = None
         self._dataloader_iter: Optional[Iterator] = None
         self._data_fetcher: Optional[DataFetcher] = None
-        self._dataloader_state_dict: Dict[str, Any] = None
+        self._dataloader_state_dict: Dict[str, Any] = {}
 
     @property
     def done(self) -> bool:
         """Returns ``True`` if the current iteration count reaches the number of dataloader batches."""
         return self.batch_progress.current.completed >= self._dl_max_batches
 
-    def connect(self, **kwargs: "Loop") -> None:
-        raise NotImplementedError(f"{self.__class__.__name__} does not connect any child loops.")
```

Comment:
this is here intentionally! otherwise eval_loop.connect(loop) will do nothing

Reply:
This is here accidentally.

1. No other leaf loops override `connect` with the `NotImplementedError`.

2. The base `Loop.connect` does nothing, if we wanted the `NotImplementedError` behaviour, it should be in the base.

3. Erroring out here makes #10324 harder, as we would need to check whether a loop has any children loop before calling `connect`.

Reply:
Fine. Remove it. With Loop.replace the connect method gets removed anyway